src/lighteval/evaluator.py

```diff
@@ -113,10 +115,24 @@ def evaluate(  # noqa: C901
             doc.instruction = ""
 
         # using a deep copy here because process results pops from the model responses
-        metrics = task.process_results(doc, copy.deepcopy(model_responses))
+        metrics = task.process_results(doc, copy.deepcopy(model_responses), evaluation_tracker=evaluation_tracker)
+
+        # Remove the user_prompt from the metrics in case of llm-as-judge metric
```

Comment:
This is not a neat system and will need to be cleaned in the future. Can we use formatted doc to store these vars?

Reply:
I agree, I will try to log it in the doc.

Reply:
I think we should be passing it as reference - if that's not the case, I'm ok will all metrics functions returning an edited formatted_doc for ex, so we get a cleaner system